### ISSUE TITLE
Game won't start with versions above 0.8.3

### ISSUE THREAD

**Issue by Nashekni** [Lyall/SMTVFix#23]
Thank you for this amazing fix, but i tried updating it from version 0.8.3 to 0.8.6 and the game won't start. The log is created, but is empty. Tried with 0.8.4 and 0.8.5 and the game won't start with those too. Any idea to what may be causing this? I really want to try the GTAO and better LoD distance.

**Comment by agentoranger** [comment]
This issue is unique to your situation, the latest version is working for me, a lowly user on the internet. Be certain that are replacing the configuration file, along with all of the with the files with the latest release, try enabling a single option at a time until the issue is recreated.

**Comment by Lyall** [comment]
All of what @agentoranger said is good advice there. You're not trying to use the fix on the Microsoft/Xbox Store version are you? There is know incompatibility with that version.

**Comment by Nashekni** [comment]
> All of what @agentoranger said is good advice there. You're not trying to use the fix on the Microsoft/Xbox Store version are you? There is known incompatibility with that version.

No, i'm using the Steam version. I just got home, so i'll try what Ranger said.

**Comment by Nashekni** [comment]
> This issue is unique to your situation, the latest version is working for me, (me being a lowly user on the internet). Be certain that you are replacing the configuration file, along with all of the with the files with the latest release, try enabling a single option at a time until the issue is recreated. Which might isolate the issue.
> 
> Supplemental info could be helpful, which video card and driver version are you using?

Hi, just started my pc and tried setting everything to false in the configuration file to see start seeing what may cause the problem, but even if everything is set to false the game still won't start with 0.8.6 fix files.
I'm using a AMD RX 6600 graphics card with Win11 Adrenaline 24.5.1 drivers.
I'm thinking maybe redownloading the game might fix it? I'll try to record a video of the problem if it helps.

**Comment by agentoranger** [comment]
Just in case it's relevant, I'm using Windows 11, Nvidia 4090, driver version 555.

**Comment by RaygeSol** [comment]
I just started having issues with 8.8. But I'm also using Special K.

**Comment by Nashekni** [comment]
Tried redownloading the game, but it still won't start with versions above 0.8.3. Tried the recent 0.8.8 and won't start with it either. Will upload a video showing the problem.

**Comment by Nashekni** [comment]
https://drive.google.com/file/d/1GTqX3wtIBAhG90g-tGUKxaCe_7R0UUVP/view?usp=sharing

Here is the video showing the game with no fix, 0.8.8 and 0.8.3 (it's on drive since is 60mb, can't upload it here)

**Comment by Lyall** [comment]
So the log file is created but empty? Can you try creating a file called `dsound.ini` and adding this to it:
```ini 
[GlobalSets]
DontLoadFromDllMain=0
```
Then try launching the game again. If that doesn't work, can you try renaming both `dsound.dll` and `dsound.ini` to `winmm.dll` and `winmm.ini` respectively. If none of that works, I can try creating a build that has parts pulled out to try and narrow down what part is causing the crash.

**Comment by Nashekni** [comment]
> So the log file is created but empty? Can you try creating a file called `dsound.ini` (in the game folder next to SMTVFix.asi/dsound.dll) and adding this to it:
> 
> ```ini
> [GlobalSets]
> DontLoadFromDllMain=0
> ```
> 
> Then try launching the game again. If that doesn't work, can you try renaming both `dsound.dll` and `dsound.ini` to `winmm.dll` and `winmm.ini` respectively. If none of that works, I can try creating a build that has parts pulled out to try and narrow down what part is causing the crash.

Created the dsound.dll and changing the names and didn't work. Would love if you create a build to see what could be the problem.

**Comment by Lyall** [comment]
Alright, this build basically has everything stripped out. It only interacts with the log file and the ini file. Give this a try and see if it generates a proper log file.
[SMTVFix_build1.zip](https://github.com/user-attachments/files/15907045/SMTVFix_build1.zip)

**Comment by Nashekni** [comment]
Replaced the 0.8.88 build with the one you send, the game now starts and does in fact generate a log file.
[SMTVFix.log](https://github.com/user-attachments/files/15907127/SMTVFix.log)

**Comment by Lyall** [comment]
> Replaced the 0.8.8 build with the one you send, the game now starts and does in fact generate a log file. [SMTVFix.log](https://github.com/user-attachments/files/15907127/SMTVFix.log)

Alright, I'll add more back in then and we'll see how that goes.

**Comment by Lyall** [comment]
This build has just the resolution hook, aspect ratio and FOV.

SMTVFix_build2.zip](https://github.com/user-attachments/files/15907196/SMTVFix_build2.zip)

**Comment by Lyall** [comment]
[SMTVFix_build3.zip](https://github.com/user-attachments/files/15907214/SMTVFix_build3.zip)

**Comment by Nashekni** [comment]
This one works too
[SMTVFix.log](https://github.com/user-attachments/files/15907246/SMTVFix.log)

**Comment by Lyall** [comment]
[SMTVFix_build4.zip](https://github.com/user-attachments/files/15907266/SMTVFix_build4.zip)

**Comment by Nashekni** [comment]
also working
[SMTVFix.log](https://github.com/user-attachments/files/15907289/SMTVFix.log)

**Comment by Lyall** [comment]
Alright this build has about 90% of the original stuff in it. I suspect this is the one that won't work, or at least that's what I'll put my bet on.

[SMTVFix_build5.zip](https://github.com/user-attachments/files/15907308/SMTVFix_build5.zip)

**Comment by Nashekni** [comment]
This one suprisingly does work.
[SMTVFix.log](https://github.com/user-attachments/files/15907327/SMTVFix.log)

**Comment by Lyall** [comment]
Alright, so that last build was only missing the intro skip. So if this build crashes then we know the issue is with that. If you can confirm that I'll see if I can add some extra checks to the intro skip to maybe fix whatever is happening on your end.

[SMTVFix_build6.zip](https://github.com/user-attachments/files/15907337/SMTVFix_build6.zip)

**Comment by Nashekni** [comment]
Ok, sooo, this one does actually starts too.
[SMTVFix.log](https://github.com/user-attachments/files/15907356/SMTVFix.log)

**Comment by Lyall** [comment]
> Ok, sooo, this one does actually starts too. [SMTVFix.log](https://github.com/user-attachments/files/15907356/SMTVFix.log)

Well now I'm confused. The only difference between the last build I sent and v0.8.8 is a few new additions. I guess it's also built through the GitHub runner versus the file I posted being compiled on my machine. I'll generate a build on GitHub of the latest version and post it here in a minute.

**Comment by Lyall** [comment]
[SMTVFix_v0.8.9.zip](https://github.com/user-attachments/files/15907413/SMTVFix_v0.8.9.zip)

**Comment by Nashekni** [comment]
This one crashed. It seems that when the fix is compiled on Github it crashes my game. Maybe my computer is missing something?

**Comment by Lyall** [comment]
> This one crashed. It seems that when the fix is compiled on Github it crashes my game. Maybe my computer is missing something?

Alright well at least we've narrowed it down now. I'd recommend installing [this pack](https://www.techpowerup.com/download/visual-c-redistributable-runtime-package-all-in-one/) of visual C++ runtimes.

**Comment by Nashekni** [comment]
Okay, I will download it right now and then i'll tell you if it worked. Thank you so much for your help and time.

**Comment by Nashekni** [comment]
> > This one crashed. It seems that when the fix is compiled on Github it crashes my game. Maybe my computer is missing something?
> 
> Alright well at least we've narrowed it down now. I'd recommend installing [this pack](https://www.techpowerup.com/download/visual-c-redistributable-runtime-package-all-in-one/) of visual C++ runtimes. Then give the GitHub-built version I posted a try.

Mmm, even after installing the c++ runtimes the game won't start with the Github built

**Comment by Lyall** [comment]
Does this build work for you. This is compiled on my machine. I've updated to the latest build tools to match the ones on GitHub's runner. The build diag output (at least at the start) is exactly the same and the filesize is a match.

[SMTVFix.zip](https://github.com/user-attachments/files/15907764/SMTVFix.zip)

**Comment by Nashekni** [comment]
> Does this build work for you? This is compiled on my machine. I've updated to the latest build tools to match the ones on GitHub's runner. The build diag output (at least at the start) is exactly the same and the file size is a match.
> 
> [SMTVFix.zip](https://github.com/user-attachments/files/15907764/SMTVFix.zip)

This build also seems to crash my game :/

**Comment by jangyoking** [comment]
> So the log file is created but empty? Can you try creating a file called `dsound.ini` (in the game folder next to SMTVFix.asi/dsound.dll) and adding this to it:
> 
> ```ini
> [GlobalSets]
> DontLoadFromDllMain=0
> ```
> 
> Then try launching the game again. If that doesn't work, can you try renaming both `dsound.dll` and `dsound.ini` to `winmm.dll` and `winmm.ini` respectively. If none of that works, I can try creating a build that has parts pulled out to try and narrow down what part is causing the crash.

Im also have problem to let it work.log files is not created.
then I rename dsound.dll to winmm.dll,it seems work

**Comment by Lyall** [comment]
> I also have problem to let it work.log files is not created. then I rename dsound.dll to winmm.dll,it seems to work

That didn't work for you right @Nashekni? I'm struggling to think why just updating the Visual Studio build tools would cause the binary to crash on your machine. I'm honestly flummoxed.

**Comment by Nashekni** [comment]
Yeah, chaging the names didn't work for me. I'm just as confused as you, like, is fully my computers problem, but I don't really know what may be causing this, like, my Win11 is fully updated and I installed the C++ things yesterday, i'll try later in the day to see what may be causing the game to crash.

**Comment by Lyall** [comment]
Well at least for now you could use the build in [this post(https://github.com/Lyall/SMTVFix/issues/23#issuecomment-2179665543). This a very baffling issue and I'm sorry there's no easy fix.

**Comment by Nashekni** [comment]
Don't worry about it, the solution probably will appear at some moment. But I really gotta thank you for everything you've done for me.

**Comment by Kierek** [comment]
I have the same issue, but I’m using Linux and Proton 9.0-2. Version 0.8.3 works great for me, so it’s not a major problem, but the changelog for newer versions looks tempting.

**Comment by Lyall** [comment]
> I have the same issue, but I’m using Linux and Proton 9.0-2. Version 0.8.3 works great for me, so it’s not a major problem, but the changelog for newer versions looks tempting.

Can you try the build from [this post](https://github.com/Lyall/SMTVFix/issues/23#issuecomment-2179665543)? This should be identical to v0.8.9, just compiled with the older build tools.

**Comment by RaygeSol** [comment]
Same thing is happening to me, but anything after 0.8.6 has stopped working for me. Used the batch to install all the runtimes and still have issues.

**Comment by fluxehub** [comment]
> I have the same issue, but I’m using Linux and Proton 9.0-2. Version 0.8.3 works great for me, so it’s not a major problem, but the changelog for newer versions looks tempting.

Issue exists for me too on Whisky on macOS (which is also using Wine), except it's only on version 0.8.7 and above, <=0.8.6 works without issue. Can also confirm that the issue doesn't occur for me on my Windows 11 desktop. What a strange one!

**Comment by lucasmathmendes** [comment]
I had this problem too. I'm on Windows 11 with GTX 4080 and the monitor is a 32:9 5120x1440. Started with version 0.8.6 a few days ago and today it started to crash, I didn't changed anything (maybe a Windows or game update caused it?). 
So I've tried to update to 0.8.9 to solve it, but it doesn't work. 
Finally, using SMTVFix_build6.zip from this post the game launched.

**Comment by Lyall** [comment]
For those that are having crashing, can you please try this build? 

[SMTVFix_v_0_8_9_buildtools.zip](https://github.com/user-attachments/files/15921859/SMTVFix_v_0_8_9_buildtools.zip)

**Comment by SuperROBO1** [comment]
> For those that are having crashing, can you please try this build?
> 
> [SMTVFix_v_0_8_9_buildtools.zip](https://github.com/user-attachments/files/15921859/SMTVFix_v_0_8_9_buildtools.zip)

had the same problem as OP and others in this thread, windows 11 2070 super ryzen 3700x, this build is working for me

**Comment by Kierek** [comment]
> For those that are having crashing, can you please try this build?
> 
> [SMTVFix_v_0_8_9_buildtools.zip](https://github.com/user-attachments/files/15921859/SMTVFix_v_0_8_9_buildtools.zip)

SMTVFix_v_0_8_9_buildtools.zip and SMTVFix_build6.zip doesn't work for me on Linux

**Comment by ASD10040** [comment]
> For those that are having crashing, can you please try this build?
> 
> [SMTVFix_v_0_8_9_buildtools.zip](https://github.com/user-attachments/files/15921859/SMTVFix_v_0_8_9_buildtools.zip)

Hi, I can confirm that this works for me too. The fix wasn't running for me suddenly when I played yesterday even when i used the new ver. So I removed it and played normally. I used this file you provided and the game launched fine. I am on Windows 10 btw.

Thanks for making this patch

**Comment by fluxehub** [comment]
> For those that are having crashing, can you please try this build?
> 
> [SMTVFix_v_0_8_9_buildtools.zip](https://github.com/user-attachments/files/15921859/SMTVFix_v_0_8_9_buildtools.zip)

Doesn't work for me on Whisky sadly

**Comment by RaygeSol** [comment]
> For those that are having crashing, can you please try this build?
> 
> [SMTVFix_v_0_8_9_buildtools.zip](https://github.com/user-attachments/files/15921859/SMTVFix_v_0_8_9_buildtools.zip)

Can confirm that this works for me on Win11.

**Comment by Lyall** [comment]
Okay so mixed results then. I've put out v0.9.0 with more build optimisations removed. If you still encounter crashing, please try changing the ASI loader filename `dsound.dll` to `d3d12.dll` or `d3d11.dll` depending on what renderer you are using. Also try adding an ini file named `d3d12.ini` or `d3d11.ini` with the contents:
``` ini
[GlobalSets]
DontLoadFromDllMain=0
```

If you are using Linux/Steam Deck, you'll need to change the launch command to reflect the new DLL name.

**Comment by agentoranger** [comment]
I'm gonna throw my hat in the ring, I use this installer for vc++ runtimes, the author (abodi) is very diligent in providing the latest vc++ runtimes libraries possible, which means they may be newer then the offical pack from microsoft. This also comes with the traditional disclaimer, your mileage may vary, and sometimes these libraries need to be removed to be updated.  Naturally, when these libraries are not present or broken common apps may begin to fail until they are reinstalled.  Example, Adobe reader needs vc++ runtimes.

Abodi's vc++ pack
https://github.com/abbodi1406/vcredist

Those looking to keep it official, go here:
https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170

**Comment by Nashekni** [comment]
> I'm gonna throw my hat in the ring, I use the abodi installer for vc++ runtimes, the author (abodi) is very diligent in providing the latest vc++ runtimes libraries possible, which means they may be newer than the official pack from microsoft, I believe they are sourced from Visual Studio. This also comes with the traditional disclaimer, your mileage may vary, and sometimes these libraries need to be removed to be updated. Sometimes they might break if the stars are not aligned. Naturally, when these libraries are not present or broken common apps may begin to fail until they are reinstalled. Example, Adobe reader needs vc++ runtimes.
> 
> Abodi's vc++ pack https://github.com/abbodi1406/vcredist
> 
> Those looking to keep it official, go here: https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170
> 
> Edit: v.0.8.9 and v.0.9.0 is working for me, Windows 11 with and without Special K.

Woah, this pack fixed the issue in my computer, now fix v.0.9.0 works without any issues. Thanks a lot!

**Comment by yakumomo011** [comment]
0.8.9 and 0.9.0 crashes the game randomly for me, tried troubleshooting with all the helpful tips here. Only 0.8.6 works for me.

**Comment by fluxehub** [comment]
> Okay so mixed results then. I've put out v0.9.0 with more build optimisations removed. If you still encounter crashing, please try changing the ASI loader filename `dsound.dll` to `d3d12.dll` or `d3d11.dll` depending on what renderer you are using. Also try adding an ini file named `d3d12.ini` or `d3d11.ini` with the contents:
> 
> ```ini
> [GlobalSets]
> DontLoadFromDllMain=0
> ```
> 
> If you are using Linux/Steam Deck, you'll need to change the launch command to reflect the new DLL name.

Still not working for me on Whisky and 0.9, tried `dsound.dll`, `d3d12.dll` and `d3d11.dll` with the respective renderers, and with the extra `.ini` file added and removed. All result in the same crash.

**Comment by Messaika** [comment]
Tried all the fixes in here, and the only thing that works outside .83 is .88 with the build1 fix. Have windows 11, Ryzen 5 3600x, ROG Strix X570F, and a RTX 3080.

**Comment by Lyall** [comment]
For those that are still having issues with crashing on startup, can you please try [v0.9.1](https://github.com/Lyall/SMTVFix/releases/tag/v0.9.1)? 

There are some changes there that may resolve the issue. For one, the C++ runtime is now statically linked meaning that you shouldn't have to worry about having the right version of visual C++ runtimes installed. The second part is an injection delay parameter. You can increase this to try and delay when hooks are being added, this value may make a difference for certain users but you'll need to experiment.

**Comment by fluxehub** [comment]
0.9.1 works for me on macOS Whisky as long as I set the delay to 10 seconds! (there may be a lower working value but I didn't test) Everything still works as expected on my PC.

**Comment by Kierek** [comment]
0.9.1 works for me too(Linux), but with default delay. Thank you!

**Comment by Messaika** [comment]
Wooo! It works with no tampering now! Thanks~

**Comment by Lyall** [comment]
Alright it seems like this is fixed for most users now in v0.9.1. There's still possibly issues with the MS Store version but that's for another issue.
